函数的调用如果出错，一般都会返回一个error对象，error是go内置的接口。

# 错误的处理策略

错误的5种处理策略

1. 向上传递，交给main处理
2. 重试一定次数，全部失败报错
3. 打印错误日志，结束程序，除遇到bug外，一般只有main可结束程序
4. 只打印错误日志
5. 忽略错误

# 文件结尾错误EOF

```go
in := bufio.NewReader(os.Stdin)
for {
	r, _, err := in.ReadRune()
    if err == io.EOF {
		break // finished reading
    }
	if err != nil {
		return fmt.Errorf("read failed:%v", err)
	}
	// ...use r…
}
```

